src/Core/PluginCollection.php

```diff
@@ -206,11 +206,42 @@ public function has(string $name): bool
      */
     public function get(string $name): PluginInterface
     {
-        if (!$this->has($name)) {
-            throw new MissingPluginException(['plugin' => $name]);
+        if ($this->has($name)) {
+            return $this->plugins[$name];
         }
 
-        return $this->plugins[$name];
+        $plugin = $this->create($name);
+        $this->add($plugin);
```

Comment:
The docblock should mention that if the plugin hasn't already been loaded it will be loaded on first access and that plugins loaded this way may miss some hook methods.

Reply:
Should `Plugin::loaded()` also autoload any installed plugins?

Reply:
We can add an argument to `loaded()` to autoload available plugins but it shouldn't be default.

Autoloading a plugin could have unknown and undesirable effects on the applications working.